themes/theme-gmd/pages/Scanner/index.jsx

```diff
@@ -42,17 +48,31 @@ class ScannerView extends PureComponent {
       <View background={backgroundColor}>
         <RouteContext.Consumer>
           {({ query: { scope = SCANNER_SCOPE_DEFAULT, type = SCANNER_TYPE_BARCODE } = {} }) => (
-            <Fragment>
+            <ScannerContainer
+              scope={scope}
+              type={type}
+              scannerDidOpen={this.removeBackground}
+              scannerDidClose={this.resetBackground}
+            >
               <BackBar title="titles.scanner" right={null} />
-              <ScannerContainer
-                scope={scope}
-                type={type}
-                scannerDidOpen={this.removeBackground}
-                scannerDidClose={this.resetBackground}
-              >
-                <ScannerOverlay />
-              </ScannerContainer>
-            </Fragment>
+              <ScannerOverlay
+                cameraOverlay={
+                  <SurroundPortals portalName={scannerPortals.SCANNER_CAMERA}>
+                    <CameraOverlay />
```

Comment:
The portals that surround the CameraOverlay should actually be INSIDE the CameraOverlay component for performance reasons. If the Scanner re-renders, they will also be re-concealed. If they are INSIDE the CameraOverlay component, they won't even be touched.